functions/api/transactions.js

```diff
@@ -7,6 +7,23 @@ const router = express.Router();
 const url = require('url');
 const db = admin.firestore();
 
+
+
+async function hasPreviousPage (transactionCollection, snapshot){
+    console.log("has function")
```

Comment:
so if you want to use .then(somevalue) -> somevalue must be returned/resolved. 
i would suggest to wrap this logic in a promise for that. 
`async function hasPreviousPage (transactionCollection, snapshot){
return new Promise(resolve,reject) {
..fn logic here
resolve(hasPreviousPage)
}
}`